modules/clickhouse/zookeeper.go

```diff
@@ -0,0 +1,129 @@
+package clickhouse
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const defaultZKImage = "zookeeper:3.7"
+
+const zkPort = nat.Port("2181/tcp")
+
+// ClickHouseContainer represents the ClickHouse container type used in the module
+type ZookeeperContainer struct {
+	testcontainers.Container
+	exposedPort nat.Port
+	ipaddr      string
+}
+
+func (c *ZookeeperContainer) IP() string {
+	return c.ipaddr
+}
+
+func (c *ZookeeperContainer) Port() nat.Port {
+	return c.exposedPort
+}
+
+func (c *ZookeeperContainer) ClickHouseConfig() string {
+	return fmt.Sprintf(`<?xml version="1.0"?>
+<clickhouse>
+    <logger>
+        <level>debug</level>
+        <console>true</console>
+        <log remove="remove"/>
+        <errorlog remove="remove"/>
+    </logger>
+
+    <query_log>
+        <database>system</database>
+        <table>query_log</table>
+    </query_log>
+
+    <timezone>Europe/Berlin</timezone>
+
+    <zookeeper>
+        <node index="1">
+            <host>%s</host>
+            <port>2181</port>
+        </node>
+    </zookeeper>
+
+    <remote_servers>
+        <default>
+            <shard>
+                <replica>
+                    <host>localhost</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </default>
+    </remote_servers>
+    <macros>
+        <cluster>default</cluster>
+        <shard>shard</shard>
+        <replica>replica</replica>
+    </macros>
+
+    <distributed_ddl>
+        <path>/clickhouse/task_queue/ddl</path>
+    </distributed_ddl>
+
+    <format_schema_path>/var/lib/clickhouse/format_schemas/</format_schema_path>
+</clickhouse>
+`, c.IP())
+}
+
+func RunZookeeper(ctx context.Context) (*ZookeeperContainer, error) {
+	zkcontainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			ExposedPorts: []string{zkPort.Port()},
+			Image:        defaultZKImage,
+			WaitingFor:   wait.ForListeningPort(zkPort),
+		},
+		Started: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	zkExposedPort, err := zkcontainer.MappedPort(ctx, zkPort)
+	if err != nil {
+		return nil, err
+	}
+
+	ipaddr, err := zkcontainer.ContainerIP(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &ZookeeperContainer{Container: zkcontainer, exposedPort: zkExposedPort, ipaddr: ipaddr}, nil
+}
```

Comment:
I don't think the clickhouse module should have a zookeeper implementation but rather just receive the connection string to be used in clickhouse

Reply:
I can move zk container into a separate module, but it's like 20 lines of code, not sure regards it worth a hassle.

Reply:
I can move this code just in examples